File: labs/07/src/Style/Style.php
<?php

namespace App\Style;

class Style implements StyleInterface
{
    private bool $enabled = true;
    private RGBAColor $color;

    public function __construct(RGBAColor $color)
    {
        $this->color = clone $color;
    }

    public function isEnabled(): bool
    {
        return $this->enabled;
    }

    public function enable(bool $enable): void
    {
        $this->enabled = $enable;
    }

    public function getColor(): RGBAColor
    {
        return $this->color;
    }

    public function setColor(RGBAColor $color): void
    {
        $this->color = clone $color;
    }
}

File: labs/09/lab9v2/lab9v2/src/style/Style.java
package style;

import java.awt.*;

public class Style implements StyleInterface {
    private Color color;

    public Style(Color color) {
        this.setColor(color);
    }

    @Override
    public Color getColor() {
        return this.color;
    }

    @Override
    public void setColor(Color color) {
        this.color = new Color(color.getRed(), color.getGreen(), color.getBlue(), color.getAlpha());
    }
}
